subprojects/groovy-parser-antlr4/src/main/java/org/apache/groovy/parser/antlr4/TryWithResourcesASTTransformation.java

```diff
@@ -342,7 +348,7 @@ private ExpressionStatement createAddSuppressedStatement(String primaryExcName,
                 new MethodCallExpression(
                         new VariableExpression(primaryExcName),
                         "addSuppressed",
-                        new ArgumentListExpression(Collections.singletonList(new VariableExpression(suppressedExcName))));
+                        new ArgumentListExpression(Collections.singletonList((Expression) new VariableExpression(suppressedExcName))));
```

Comment:
True! But the type inference on Java 7 needs this, or you'll get a type mismatch, since a List<VariableExpression> is not the same as List<Expression>. On Java 8, the type inference can solve this by itself, by checking the supertypes of the generic argument to singletonList.

I found 5-6 of these kind of required change.